### DISCUSSION TITLE
Arquillian Remote Domain Container With HTTPS Only Configured On Wildfly Server

### DISCUSSION THREAD

**Discussion by fearghalom** [wildfly/wildfly-arquillian#234] [Q&A]
I'm wondering is it possible to get Arquillian tests to run with a remote domain container when the http listener has been disabled on the Wildfly instance under test? When I enable the http-listener, the tests execute successfully. 
I debugged through the Arquillian code and noticed that the protocol seems to be hardcoded to HTTP in org.jboss.as.arquillian.container.domain.ManagementClient when determining the subsystem URI.

**Comment by jamezp** [comment]
The management HTTP does need to be listening. There is a in 4.0.0.Alpha5 and 5.0.0.Alpha5 to use a different binding name https://issues.redhat.com/browse/WFARQ-31. You can add the `socketBindingGroup` property to your `arquillian.xml`.

**Reply by fearghalom** [reply]
thanks James, appreciate the prompt response. Let me try upgrading to one of the versions you've listed.

**Reply by fearghalom** [reply]
I'm using the Remote Domain Container adapter (wildfly-arquillian-container-domain-remote). Have the changes to support a different binding name made it into this?

I upgraded to v4.0.0.Alpha5 from v2.2.0.Final. I noticed that for my test, it's the org.jboss.as.arquillian.container.domain.ManagementClient that get's hit and this doesn't seem to have been updated in PR: https://github.com/wildfly/wildfly-arquillian/pull/228 .

I also tried upgrading to v5.0.0.Alpha5 but I noticed that the dependency injections in my tests stopped working.

**Reply by jamezp** [reply]
Sorry for the late reply. It does look like domain was skipped. I forgot that domain did not rely on the main common module. I've filed https://issues.redhat.com/browse/WFARQ-120 for that. I'll try to get to it next week.

**Reply by jamezp** [reply]
I've resolved the JIRA and added a `socketBindingName` and `protocol` option for domain containers. I'll probably do a release next week which includes this.

**Reply by fearghalom** [reply]
That's great,  thanks James for addressing this so promptly

**Reply by jamezp** [reply]
I forgot to mention last week that [WildFly Arquillian 4.0.0.Alpha6](https://github.com/wildfly/wildfly-arquillian/releases/tag/4.0.0.Alpha6) for Jakarta EE 8 and [WildFly Arquillian 5.0.0.Alpha6](https://github.com/wildfly/wildfly-arquillian/releases/tag/5.0.0.Alpha6) for Jakarta EE 9/10 was released. It includes the support for overriding the socket-binding in domain.

The property name was changed to `socketBindingName`. Also 5.0.0.Alpha6 now requires Java 11.

**Reply by fearghalom** [reply]
I upgraded to v4.0.0.Alpha6 and configured the arquillian.xml as follows:

```
<?xml version="1.0" encoding="UTF-8"?>

<arquillian xmlns="http://jboss.org/schema/arquillian"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://jboss.org/schema/arquillian http://jboss.org/schema/arquillian/arquillian_1_0.xsd">

    <container qualifier="jboss" default="true">
        <configuration>
            <property name="socketBindingName">https</property>
            <property name="protocol">https</property>
        </configuration>
    </container>
</arquillian>
```

However, I'm getting the following exception now:

```
java.net.SocketException: Unexpected end of file from server
	at java.base/sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:862)
	at java.base/sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:689)
	at java.base/sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:859)
	at java.base/sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:689)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1615)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1520)
	at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:527)
	at org.jboss.arquillian.protocol.servlet.ServletMethodExecutor.execute(ServletMethodExecutor.java:168)
	at org.jboss.arquillian.protocol.servlet.ServletMethodExecutor$1.run(ServletMethodExecutor.java:221)
	at java.base/java.util.TimerThread.mainLoop(Timer.java:556)
	at java.base/java.util.TimerThread.run(Timer.java:506)
```

Is there more I need to configure to get this to work? Does the servlet protocol support authentication?

**Reply by jamezp** [reply]
I don't think it handles authentication, no. What kind of authentication would you want to support? It's something to be considered for sure.

**Reply by fearghalom** [reply]
Basic authentication would do to begin with.
